book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/repository/EventRepository.java

```diff
@@ -0,0 +1,17 @@
+package org.lbcc.bms.bms_monolith.repository;
+
+import org.lbcc.bms.bms_monolith.entity.Event;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.UUID;
+
+@Repository
+public interface EventRepository extends JpaRepository<Event, UUID> {
+
+    @Query("SELECT e FROM Event e JOIN FETCH e.vendor JOIN FETCH e.venue JOIN FETCH e.eventType")
```

Comment:
I think this is not required. Bydefault JPA will use Joins. We can test it by running the code and checking hibernate logs.

Reply:
Right, it seems there is an issue with the entity mapping, resulting in an entity not found exception